cmd/launcher/uninstall_linux.go

```diff
@@ -0,0 +1,85 @@
+//go:build linux
+// +build linux
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/pkg/errors"
+)
+
+func removeLauncher(ctx context.Context, logger log.Logger, identifier string) error {
+	if identifier == "" {
+		identifier = "kolide-k2"
+	}
+
+	isDebianOS, err := isDebian(ctx, logger)
+	if err != nil {
+		return err
+	}
+
+	serviceName := fmt.Sprintf("launcher.%s", identifier)
+	packageName := fmt.Sprintf("launcher-%s", identifier)
+
+	// Stop launcher service
+	if _, err := tablehelpers.Exec(ctx, logger, 30, []string{"systemctl"}, []string{"stop", serviceName}); err != nil {
+		return err
+	}
+
+	// Disable launcher service
+	if _, err := tablehelpers.Exec(ctx, logger, 30, []string{"systemctl"}, []string{"disable", serviceName}); err != nil {
+		return err
+	}
+
+	// Tell the appropriate package manager to remove launcher
+	if isDebianOS {
+		if _, err := tablehelpers.Exec(ctx, logger, 30, []string{"dpkg"}, []string{"--purge", packageName}); err != nil {
+			return err
+		}
+	} else {
+		if _, err := tablehelpers.Exec(ctx, logger, 30, []string{"rpm"}, []string{"-e", packageName}); err != nil {
+			return err
+		}
+	}
+
+	dirsToRemove := []string{
+		fmt.Sprintf("/var/%s", identifier),
+		fmt.Sprintf("/etc/%s", identifier),
+	}
+
+	// Now remove the local dirs used for launcher/osquery binaries and app data
+	for _, dir := range dirsToRemove {
+		if err := os.RemoveAll(dir); err != nil {
+			fmt.Printf("error removing dir %s: %s\n", dir, err)
+		}
+	}
```

Comment:
I am scared of this accidentally removing something it shouldn't. (Like if identifier is nil)

Thinking aloud, what ways can we ensure it's safer. We can check the paths. We can display them to the user with confirmation.

Thoughts?

Reply:
Yeah I'm scared too - I put in a protection in the code to prevent identifier being an empty string, but that doesn't feel adequate.

User confirmation feels better - but maybe needs a bypass to allow automated uninstalls?

To avoid this all together, the packaging layer could generate these paths or whole commands in a file, which we just read and process.

Reply:
I've changed the identifier extraction to be via regexp for a better guarantee. We can further restrict the regexp based on what are acceptable identifier chars.

Reply:
Still thinking about this one.